src/i18n.tsx

```diff
@@ -82,21 +83,24 @@ async function dynamicActivate(locale: SupportedLocale) {
   i18n.activate(locale)
 }
 
+dynamicActivate(initialLocale)
+
 export function LanguageProvider({ children }: { children: ReactNode }) {
-  useSetLocaleFromUrl()
   const locale = useActiveLocale()
+  const [, setUserLocale] = useUserLocaleManager()
   const [loaded, setLoaded] = useState(false)
 
   useEffect(() => {
     dynamicActivate(locale)
       .then(() => {
         document.documentElement.setAttribute('lang', locale)
+        setUserLocale(locale) // stores the selected locale to persist across sessions
         setLoaded(true)
       })
       .catch((error) => {
         console.error('Failed to activate locale', locale, error)
       })
-  }, [locale])
+  }, [locale, setUserLocale])
 
   // prevent the app from rendering with placeholder text before the locale is loaded
```

Comment:
do we still need this bit?

Reply:
yes - in the case that the locale is not the default/fallback locale, this prevents the extra network request for the fallback if the locale is not yet loaded.